docs/content.md

```diff
@@ -0,0 +1,198 @@
+- **Safe publishing** – publishing promotes the draft version to live content, while revert snaps back to the last publication. Hidden blocks stay out of the frontend until re-enabled.
+
+## Workflow: Adding Content To A Module (Pages Example)
+This walkthrough assumes you already have a `Page` model with title/slug attributes and an admin module scaffolded in Koi.
+
+1. **Install migrations**
+   - run `bin/rails katalyst_content:install:migrations`
+   - apply the generated migrations plus the Page-specific version table:
+     ```ruby
+     class CreatePageVersions < ActiveRecord::Migration[7.0]
+       def change
+         create_table :page_versions do |t|
+           t.references :parent, null: false, foreign_key: { to_table: :pages }
+           t.json :nodes
+           t.timestamps
+         end
+
+         change_table :pages do |t|
+           t.references :published_version, foreign_key: { to_table: :page_versions }
+           t.references :draft_version,     foreign_key: { to_table: :page_versions }
+         end
+       end
+     end
+     ```
+   - **Verification:** `bin/rails db:migrate` succeeds and `rails db:schema:dump` shows `katalyst_content_items` plus `page_versions` and the extra foreign keys on `pages`.
+
+2. **Update the model**
+   ```ruby
+   # app/models/page.rb
+   class Page < ApplicationRecord
+     include Katalyst::Content::Container
+
+     class Version < ApplicationRecord
+       include Katalyst::Content::Version
+     end
+   end
+   ```
+   - **Verification:** `bin/rails console` – instantiate a page (`page = Page.new`) and set an items payload (`page.items_attributes = []`). `page.draft_version` should now return a version object, while `page.publish!` still raises because nothing is persisted yet. Assigning `items_attributes` (even an empty array) mimics the editor’s requests and is required before the container builds versions.
+
+3. **Expose content in the admin controller**
+   Adapt your admin controller to match the dummy implementation:
+   ```ruby
+   class Admin::PagesController < Admin::ApplicationController
+     helper Katalyst::Content::EditorHelper
+
+     def show
+       page   = Page.find(params[:id])
+       editor = Katalyst::Content::EditorComponent.new(container: page)
+       render locals: { page:, editor: }
+     end
+
+     def update
+       page = Page.find(params[:id])
+       page.attributes = page_params
+
+       unless page.valid?
+         editor = Katalyst::Content::EditorComponent.new(container: page)
+         respond_to do |format|
+           format.turbo_stream { render editor.errors, status: :unprocessable_entity }
+         end
+         return
+       end
+
+       case params[:commit]
+       when "publish" then page.save! && page.publish!
+       when "revert"  then page.revert!
+       else                  page.save!
+       end
+
+       redirect_to [:admin, page], status: :see_other
```

Comment:
This code all looks pretty dated (i.e. early Koi 4). Where did you find it? I'd like to provide a better example